tedana/model/combine.py

```diff
@@ -45,45 +128,47 @@ def make_optcom(data, tes, mask, t2s=None, combmode='t2s', verbose=True):
     2.  Perform weighted average per voxel and TR across TEs based on weights
         estimated in the previous step.
     """
-    if combmode == 't2s' and t2s is None:
+    if data.ndim != 3:
+        raise ValueError('Input data must be 3D (S x E x T)')
+
+    if len(tes) != data.shape[1]:
+        raise ValueError('Number of echos provided does not match second '
+                         'dimension of input data: {0} != '
+                         '{1}'.format(len(tes), data.shape[1]))
+
+    if mask.ndim != 1:
+        raise ValueError('Mask is not 1D')
+    elif mask.shape[0] != data.shape[0]:
+        raise ValueError('Mask and data do not have same number of '
+                         'voxels/samples: {0} != {1}'.format(mask.shape[0],
+                                                             data.shape[0]))
+
+    if combmode not in ['t2s', 'ste']:
+        raise ValueError("Argument 'combmode' must be either 't2s' or 'ste'")
+    elif combmode == 't2s' and t2s is None:
         raise ValueError("Argument 't2s' must be supplied if 'combmode' is "
                          "set to 't2s'.")
     elif combmode == 'ste' and t2s is not None:
         LGR.warning("Argument 't2s' is not required if 'combmode' is 'ste'.")
```

Comment:
Can we modify the warning a bit, here ? I imagine that if someone supplies `t2s` and `combmode=ste` they might have misunderstood something.

Maybe we can add "Please make check if calling arguments match input data!" or something to that effect ?

Reply:
I just realized I committed changes without responding to this. What about the following? It's informative but not very strong.

```python
LGR.warning("Argument 't2s' is not required if 'combmode' is 'ste'. "
            "'t2s' array will not be used.")
```